File: Arcade/Intro/02_Edge_of_the_Ocean/07_almostIncreasingSequence/main.py
def almostIncreasingSequence(sequence):
    s_1 = list(sequence)
    s_2 = list(sequence)
    for i in range(0, len(sequence)-1):
        if sequence[i] >= sequence[i+1]:
            s_1.pop(i)
            s_2.pop(i+1)
            break
    if s_1 == sorted(set(s_1)) and s_1 == sorted(s_1):
        return True
    elif s_2 == sorted(set(s_2)) and s_2 == sorted(s_2):
        return True
    else:
        return False

File: Arcade/Intro/02_Edge_of_the_Ocean/07_almostIncreasingSequence/main.ts
function almostIncreasingSequence(sequence: number[]): boolean {
    let sequence1: number[] = sequence.slice();
    let sequence2: number[] = sequence.slice();
    for (let i: number = 1; i < sequence.length; i++) {
        if (sequence[i-1] >= sequence[i]) {
            sequence1.splice(i-1, 1);
            sequence2.splice(i, 1);
            break;
        }
    }
    if (sequence.length === sequence2.length) return true;
    let seq1: boolean = false;
    let seq2: boolean = false;
    for (let i: number = 1; i < sequence1.length; i++) {
        if (sequence1[i-1] >= sequence1[i]) seq1 = true;
        if (sequence2[i-1] >= sequence2[i]) seq2 = true;
        if (seq1 && seq2) return false;
    }
    return true;
}
